pkg/terraform/terraform.go

```diff
@@ -0,0 +1,190 @@
+func (t *TerraformClient) Plan(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error) {
+	childArgs := []string{"plan"}
+	childArgs = append(childArgs, args...)
+
+	if file != "" {
+		childArgs = append(childArgs, fmt.Sprintf("-out=%s", file))
+	}
+
+	return child.Run(ctx, workingDir, "terraform", childArgs) //nolint:wrapcheck
+}
+
+// Apply runs the Terraform apply command.
+func (t *TerraformClient) Apply(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error) {
+	childArgs := []string{"apply"}
+	childArgs = append(childArgs, args...)
+
+	if file != "" {
+		childArgs = append(childArgs, file)
+	}
+
+	return child.Run(ctx, workingDir, "terraform", childArgs) //nolint:wrapcheck
+}
+
+// GetEntrypointDirectories gets all the directories that have Terraform config
+// files containing a backend block to be used as an entrypoint module.
+func (t *TerraformClient) GetEntrypointDirectories(rootDir string) ([]string, error) {
+	matches := make(map[string]struct{})
+	if err := filepath.WalkDir(rootDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return fmt.Errorf("failed to walk directory %s: %w", path, err)
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		matched, err := filepath.Match("*.tf", filepath.Base(path))
+		if err != nil {
+			return fmt.Errorf("failed to find terraform files: %w", err)
+		}
+
+		if !matched {
+			return nil
+		}
+
+		hasBackend, _, err := hasBackendConfig(path)
+		if err != nil {
+			return fmt.Errorf("failed to find terraform backend config: %w", err)
+		}
+
+		if !hasBackend {
+			return nil
+		}
+
+		matches[filepath.Dir(path)] = struct{}{}
+
+		return nil
+	}); err != nil {
+		return nil, fmt.Errorf("failed to find files: %w", err)
+	}
+
+	dirs := []string{}
+
+	for dir := range matches {
+		dirs = append(dirs, dir)
+	}
+
+	sort.Strings(dirs)
+
+	return dirs, nil
+}
+
+// hasBackendConfig tests a Terraform config file for the existence of a backend block.
+func hasBackendConfig(path string) (bool, hcl.Diagnostics, error) {
+	var diags hcl.Diagnostics
+
+	if _, err := os.Stat(path); err != nil {
+		return false, nil, fmt.Errorf("failed to find file: %w", err)
+	}
+
+	parser := hclparse.NewParser()
+	file, d := parser.ParseHCLFile(path)
+	diags = append(diags, d...)
+
+	rootBlocks, _, d := file.Body.PartialContent(RootSchema)
+	diags = append(diags, d...)
+
+	for _, terraformBlock := range rootBlocks.Blocks.OfType("terraform") {
+		content, _, d := terraformBlock.Body.PartialContent(TerraformSchema)
+		diags = append(diags, d...)
+
+		innerBlocks := content.Blocks.OfType("backend")
+		if len(innerBlocks) > 0 {
+			return true, diags, nil
+		}
+	}
+
+	return false, diags, nil
+}
+
+// FormatOutputForGitHubDiff formats the Terraform diff output for use with
+// GitHub diff markdown formatting.
+func FormatOutputForGitHubDiff(content string) string {
+	changed := regexp.MustCompile("(?m)^([\t ]*)([~])")
```

Comment:
It might be helpful to document this regex with comments like:

```
regexp.MustCompile(
  "(?m)" + // enable multi-line mode
  " .... "  + // tildes are special because foo
```